library/src/validations/imei/imei.ts

```diff
@@ -12,8 +12,8 @@ import { getOutput, getPipeIssues, isLuhnAlgo } from '../../utils/index.ts';
  */
 export function imei<TInput extends string>(error?: ErrorMessage) {
   return (input: TInput): PipeResult<TInput> =>
-    !/^\d{2}[ |/|-]?\d{6}[ |/|-]?\d{6}[ |/|-]?\d$/.test(input) ||
-    !isLuhnAlgo(input)
+    // eslint-disable-next-line security/detect-unsafe-regex -- false positive according to https://devina.io/redos-checker
```

Comment:
If this rule is so buggy, we should consider removing it in general or create an issue on GitHub.

Reply:
Yeah.. this is one of the longest standing issues of the security plugin: https://github.com/eslint-community/eslint-plugin-security/issues/28

Two alternatives exist:
- https://makenowjust-labs.github.io/recheck/docs/usage/as-eslint-plugin/ (discussed in that thread as well and wasn't picked because of some slow runs; this might have been fixed with the async run introduced recently?)
- https://github.com/tjenkinson/eslint-plugin-redos-detector (not mentioned in the thread)

For my personal config I've opted with sticking with eslint-plugin-security until they've sorted out what to use, but I wasn't aware of `eslint-plugin-redos-detector` until now. Maybe worth trying out. I'll also mention that one in the thread above

Reply:
As long as there is no fix, I prefer to disable or replace this linting rule.

Reply:
I'll play with the two plugins and update the PR accordingly

Reply:
https://github.com/colinhacks/zod/pull/2849 the zod PR is also interesting for this PR I'd say

Reply:
Yes we should keep an eye on this

Reply:
I decided to go with redos-detector, since it seems to find more (non-false positives) of redos regexps.